### PR TITLE
ORC-1191: Updated TLC Taxi Benchmark Dataset

### PR DESCRIPTION
**Pull request by mwlon**
### What changes were proposed in this pull request?
Update to new NYC TLC Yellow Taxi dataset

* https://www1.nyc.gov/assets/tlc/downloads/pdf/data_dictionary_trip_records_yellow.pdf
* replace non-existent CSV URLs with new Parquet URLs
* update file format in referencing benchmarks
* change util that converted from CSV string -> decimal to convert from parquet double -> decimal

### Why are the changes needed?

As of 2022-05-13, NYC TLC switched from CSVs to Parquet files. They also changed their schema slightly. Now the CSVs are no longer publicly available, so to have a replicable benchmark suite, we should refer to the new version.

### How was this patch tested?
This does not change functionality or tests (except for the updated taxi dataset).

```
java -jar hive/target/orc-benchmarks-hive-1.9.0-SNAPSHOT-uber.jar decimal data
...
# Run progress: 50.00% complete, ETA 00:00:02
# Fork: 1 of 1
# Warmup Iteration   1: [WARN ] Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
122173.181 us/op
# Warmup Iteration   2: 116629.135 us/op
Iteration   1: 118590.860 us/op
Iteration   2: 116885.746 us/op
Iteration   3: 117235.593 us/op
...
```

### CODE REVIEW

File: java/bench/core/src/java/org/apache/orc/bench/core/Driver.java [hunk 1]
```diff
@@ -18,10 +18,7 @@
 
 package org.apache.orc.bench.core;
 
-import java.util.Arrays;
-import java.util.Map;
-import java.util.ServiceLoader;
-import java.util.TreeMap;
+import java.util.*;
```
Comment (dongjoon-hyun) [inline java/bench/core/src/java/org/apache/orc/bench/core/Driver.java:21]: Please recover this.

File: java/bench/fetch-data.sh [hunk 2]
```diff
@@ -15,8 +15,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 mkdir -p data/sources/taxi
-(cd data/sources/taxi; wget -O - https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2015-11.csv | gzip > yellow_tripdata_2015-11.csv.gz )
-(cd data/sources/taxi; wget -O - https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2015-12.csv | gzip > yellow_tripdata_2015-12.csv.gz )
+(cd data/sources/taxi; wget https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2015-11.parquet )
+(cd data/sources/taxi; wget https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2015-12.parquet )
```
Comment (dongjoon-hyun) [inline java/bench/fetch-data.sh:19]: I guess there is a reason why Apache ORC didn't download `.parquet` format. ;)

File: java/mapreduce/pom.xml [hunk 3]
```diff
@@ -63,10 +63,6 @@
       <groupId>org.apache.hive</groupId>
       <artifactId>hive-storage-api</artifactId>
     </dependency>
-    <dependency>
-      <groupId>org.slf4j</groupId>
-      <artifactId>slf4j-api</artifactId>
-    </dependency>
```
Comment (dongjoon-hyun) [inline java/mapreduce/pom.xml:69]: Could you spin off this as a separate PR? We prefer to keep one theme in one PR. Especially, we avoid mixing 1) bug fix, 2) improvement, 3) clean-up. This looks like some kind of clean-up which is irrelevant to your PR intention.

File: java/pom.xml [hunk 4]
```diff
@@ -314,6 +314,7 @@
           </executions>
           <configuration>
             <failOnWarning>true</failOnWarning>
+            <ignoreNonCompile>true</ignoreNonCompile>
```
Comment (dongjoon-hyun) [inline java/pom.xml:317]: Do we need this?

File: java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java [hunk 5]
```diff
@@ -221,8 +221,8 @@ public static BatchReader createReader(Path root,
                                          long salesRecords) throws IOException {
     switch (dataName) {
       case "taxi":
-        return new RecursiveReader(new Path(root, "sources/" + dataName), "csv",
-            schema, conf, CompressionKind.ZLIB);
+        return new RecursiveReader(new Path(root, "sources/" + dataName), "parquet",
+            schema, conf, CompressionKind.NONE);
```
Comment (dongjoon-hyun) [inline java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java:225]: I believe this should be recovered along with https://github.com/apache/orc/pull/1141/files#r881201813

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for making a PR, @mwlon .

File: java/mapreduce/pom.xml (hunk 3 above)
Comment (mwlon) [reply java/mapreduce/pom.xml:69]: The maven package command needed to run the benchmarks fails without it. Should I still make it a separate PR that we merge in first?

File: java/pom.xml (hunk 4 above)
Comment (mwlon) [reply java/pom.xml:317]: I believe so - the maven dependency plugin gives warnings for test-scoped dependencies not used in main (which causes packaging to fail) without this

File: java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java (hunk 5 above)
Comment (mwlon) [reply java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java:225]: What do you mean be recovered? The CSV files are no longer available

File: java/mapreduce/pom.xml (hunk 3 above)
Comment (dongjoon-hyun) [reply java/mapreduce/pom.xml:69]: What command did you try?

**Comment by dongjoon-hyun** [comment]
BTW, I saw your comment on JIRA which is not a part of this PR description.
> NYC Taxi dataset used in benchmarks no longer exists as CSV's; has been replaced with Parquet

Actually, I used the downloaded one. So I didn't realized that. In that case, I agree with you to download Parquet data.

File: java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java (hunk 5 above)
Comment (dongjoon-hyun) [reply java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java:225]: https://github.com/apache/orc/pull/1141#issuecomment-1136712610

File: java/mapreduce/pom.xml (hunk 3 above)
Comment (mwlon) [reply java/mapreduce/pom.xml:69]: `./mvnw clean package -Pbenchmark`, as stated in the benchmark documentation

File: java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java (hunk 5 above)
Comment (dongjoon-hyun) [reply java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java:225]: I overlooked this sentence, `updated taxi dataset source and schema (the old one no longer exists)`, at the initial review.

Comment (dongjoon-hyun) [reply java/bench/core/src/java/org/apache/orc/bench/core/convert/GenerateVariants.java:225]: So, never mind about Parquet downloading part. I'll revise those comments.

File: java/bench/README.md [hunk 6]
```diff
@@ -16,7 +16,7 @@ There are three sub-modules to try to mitigate dependency hell:
 To build this library, run the following in the parent directory:
 
 ```
-% ./mvnw clean package -Pbenchmark
+% ./mvnw clean package -Pbenchmark -Dmaven.test.skip
```
Comment (dongjoon-hyun) [inline java/bench/README.md:19]: Please use `-DskipTests` if you want this.

File: java/mapreduce/pom.xml (hunk 3 above)
Comment (mwlon) [reply java/mapreduce/pom.xml]: Ah actually it looks like I was using with `-Dmaven.test.skip`, but when I changed that to `-DskipTests` like you said it was fixed. Updated.

File: java/bench/core/src/java/org/apache/orc/bench/core/Driver.java [hunk 7]
```diff
@@ -32,8 +34,14 @@ public class Driver {
...
     Map<String, OrcBenchmark> result = new TreeMap<>();
-    for(OrcBenchmark bench: loader) {
-      result.put(bench.getName(), bench);
+    Iterator<OrcBenchmark> iterator = loader.iterator();
+    while (iterator.hasNext()) {
+      try {
+        OrcBenchmark bench = iterator.next();
+        result.put(bench.getName(), bench);
+      } catch (ServiceConfigurationError e) {
+        System.out.println("WARN (Driver): misconfigured benchmark exists:" + e);
```
Comment (dongjoon-hyun) [inline java/bench/core/src/java/org/apache/orc/bench/core/Driver.java:43]: Shall we use `System.err`?
```java
System.out.println("WARN (Driver): misconfigured benchmark exists:" + e);
System.err.println("Misconfigured benchmark exists:" + e);
```

File: java/bench/core/src/java/org/apache/orc/bench/core/Driver.java [hunk 8]
```diff
@@ -32,8 +34,14 @@ public class Driver {
 
   private static Map<String, OrcBenchmark> getBenchmarks() {
     Map<String, OrcBenchmark> result = new TreeMap<>();
-    for(OrcBenchmark bench: loader) {
-      result.put(bench.getName(), bench);
+    Iterator<OrcBenchmark> iterator = loader.iterator();
```
Comment (dongjoon-hyun) [inline java/bench/core/src/java/org/apache/orc/bench/core/Driver.java:37]: Why do we need to define `iterator` variable separately?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
This PR still have 5 orthogonal themes. As you know, we need to split in order to merge, @mwlon .
- Minor `git` and `docs` changes: `java/bench/data` and `README.md`
- Adding additional exception handling in `Driver.java`.
- Update according to `tax` data change
- A typo fix in help message: `convert` -> `generate`
- Bug fix at `AvroReader` multiplier change(?)

**Comment by dongjoon-hyun** [comment]
Shall we spin off the following at least as another PR? I can help you by merging it first.
- Minor git and docs changes: java/bench/data and README.md
- A typo fix in help message: `convert` -> `generate`

Please add the following example in your new PR description.
```
$ java -jar core/target/orc-benchmarks-core-*-uber.jar generate data --help
usage: convert <root>
 -c,--compress <arg>   List of compression
 -d,--data <arg>       List of data sets
 -f,--format <arg>     List of formats
 -h,--help             Provide help
 -s,--sales <arg>      Number of records for sales
```

File: java/bench/fetch-data.sh [hunk 9]
```diff
@@ -15,8 +15,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 mkdir -p data/sources/taxi
-(cd data/sources/taxi; wget -O - https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2015-11.csv | gzip > yellow_tripdata_2015-11.csv.gz )
-(cd data/sources/taxi; wget -O - https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2015-12.csv | gzip > yellow_tripdata_2015-12.csv.gz )
+(cd data/sources/taxi; wget https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2015-11.parquet )
```
Comment (dongjoon-hyun) [inline java/bench/fetch-data.sh:18]: BTW, could you update the following accordingly because the new file are very small now?
https://github.com/apache/orc/blob/1afc31d6c04729d7e194a6423c690af4519aab33/java/bench/README.md#L27

**Comment by dongjoon-hyun** [comment]
Please rebase and use different JIRA ID because https://github.com/apache/orc/pull/1142 is merged as ORC-1189 already.

**Comment by dongjoon-hyun** [comment]
#1143 is merged.

**Comment by mwlon** [comment]
@dongjoon-hyun this is ready for review when you get a chance

**Comment by dongjoon-hyun** [comment]
Thank you. Could you update the PR title, @mwlon ?

**Comment by mwlon** [comment]
@dongjoon-hyun done.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @mwlon .

File: java/bench/README.md [hunk 10]
```diff
@@ -24,7 +24,7 @@ To fetch the source data:
 
 ```% ./fetch-data.sh```
 
-> :warning: Script will fetch 7GB of data
+> :warning: Script will fetch 500MB of data
```
Comment (dongjoon-hyun) [inline java/bench/README.md:27]: During the final verification at the merging step, I found that this is wrong. Maybe, you didn't run `fetch-data.sh` successfully?
```
$ ./fetch-data.sh
...
FINISHED --2022-05-29 14:08:32--
Total wall clock time: 7m 21s
Downloaded: 360 files, 3.5G in 4m 34s (13.2 MB/s)

$ du -h data
3.6G	data/sources/github
321M	data/sources/taxi
3.9G	data/sources
3.9G	data

$ tree data
data
└── sources
    ├── github
    │   ├── 2015-11-01-0.json.gz
    │   ├── 2015-11-01-1.json.gz
    │   ├── 2015-11-01-10.json.gz
    │   ├── 2015-11-01-11.json.gz
    │   ├── 2015-11-01-12.json.gz
    │   ├── 2015-11-01-13.json.gz
    │   ├── 2015-11-01-14.json.gz
    │   ├── 2015-11-01-15.json.gz
    │   ├── 2015-11-01-16.json.gz
    │   ├── 2015-11-01-17.json.gz
    │   ├── 2015-11-01-18.json.gz
    │   ├── 2015-11-01-19.json.gz
    │   ├── 2015-11-01-2.json.gz
    │   ├── 2015-11-01-20.json.gz
    │   ├── 2015-11-01-21.json.gz
    │   ├── 2015-11-01-22.json.gz
    │   ├── 2015-11-01-23.json.gz
    │   ├── 2015-11-01-3.json.gz
    │   ├── 2015-11-01-4.json.gz
    │   ├── 2015-11-01-5.json.gz
    │   ├── 2015-11-01-6.json.gz
    │   ├── 2015-11-01-7.json.gz
    │   ├── 2015-11-01-8.json.gz
    │   ├── 2015-11-01-9.json.gz
    │   ├── 2015-11-02-0.json.gz
    │   ├── 2015-11-02-1.json.gz
    │   ├── 2015-11-02-10.json.gz
    │   ├── 2015-11-02-11.json.gz
    │   ├── 2015-11-02-12.json.gz
    │   ├── 2015-11-02-13.json.gz
    │   ├── 2015-11-02-14.json.gz
    │   ├── 2015-11-02-15.json.gz
    │   ├── 2015-11-02-16.json.gz
    │   ├── 2015-11-02-17.json.gz
    │   ├── 2015-11-02-18.json.gz
    │   ├── 2015-11-02-19.json.gz
    │   ├── 2015-11-02-2.json.gz
    │   ├── 2015-11-02-20.json.gz
    │   ├── 2015-11-02-21.json.gz
    │   ├── 2015-11-02-22.json.gz
    │   ├── 2015-11-02-23.json.gz
    │   ├── 2015-11-02-3.json.gz
    │   ├── 2015-11-02-4.json.gz
    │   ├── 2015-11-02-5.json.gz
    │   ├── 2015-11-02-6.json.gz
    │   ├── 2015-11-02-7.json.gz
    │   ├── 2015-11-02-8.json.gz
    │   ├── 2015-11-02-9.json.gz
    │   ├── 2015-11-03-0.json.gz
    │   ├── 2015-11-03-1.json.gz
    │   ├── 2015-11-03-10.json.gz
    │   ├── 2015-11-03-11.json.gz
    │   ├── 2015-11-03-12.json.gz
    │   ├── 2015-11-03-13.json.gz
    │   ├── 2015-11-03-14.json.gz
    │   ├── 2015-11-03-15.json.gz
    │   ├── 2015-11-03-16.json.gz
    │   ├── 2015-11-03-17.json.gz
    │   ├── 2015-11-03-18.json.gz
    │   ├── 2015-11-03-19.json.gz
    │   ├── 2015-11-03-2.json.gz
    │   ├── 2015-11-03-20.json.gz
    │   ├── 2015-11-03-21.json.gz
    │   ├── 2015-11-03-22.json.gz
    │   ├── 2015-11-03-23.json.gz
    │   ├── 2015-11-03-3.json.gz
    │   ├── 2015-11-03-4.json.gz
    │   ├── 2015-11-03-5.json.gz
    │   ├── 2015-11-03-6.json.gz
    │   ├── 2015-11-03-7.json.gz
    │   ├── 2015-11-03-8.json.gz
    │   ├── 2015-11-03-9.json.gz
    │   ├── 2015-11-04-0.json.gz
    │   ├── 2015-11-04-1.json.gz
    │   ├── 2015-11-04-10.json.gz
    │   ├── 2015-11-04-11.json.gz
    │   ├── 2015-11-04-12.json.gz
    │   ├── 2015-11-04-13.json.gz
    │   ├── 2015-11-04-14.json.gz
    │   ├── 2015-11-04-15.json.gz
    │   ├── 2015-11-04-16.json.gz
    │   ├── 2015-11-04-17.json.gz
    │   ├── 2015-11-04-18.json.gz
    │   ├── 2015-11-04-19.json.gz
    │   ├── 2015-11-04-2.json.gz
    │   ├── 2015-11-04-20.json.gz
    │   ├── 2015-11-04-21.json.gz
    │   ├── 2015-11-04-22.json.gz
    │   ├── 2015-11-04-23.json.gz
    │   ├── 2015-11-04-3.json.gz
    │   ├── 2015-11-04-4.json.gz
    │   ├── 2015-11-04-5.json.gz
    │   ├── 2015-11-04-6.json.gz
    │   ├── 2015-11-04-7.json.gz
    │   ├── 2015-11-04-8.json.gz
    │   ├── 2015-11-04-9.json.gz
    │   ├── 2015-11-05-0.json.gz
    │   ├── 2015-11-05-1.json.gz
    │   ├── 2015-11-05-10.json.gz
    │   ├── 2015-11-05-11.json.gz
    │   ├── 2015-11-05-12.json.gz
    │   ├── 2015-11-05-13.json.gz
    │   ├── 2015-11-05-14.json.gz
    │   ├── 2015-11-05-15.json.gz
    │   ├── 2015-11-05-16.json.gz
    │   ├── 2015-11-05-17.json.gz
    │   ├── 2015-11-05-18.json.gz
    │   ├── 2015-11-05-19.json.gz
    │   ├── 2015-11-05-2.json.gz
    │   ├── 2015-11-05-20.json.gz
    │   ├── 2015-11-05-21.json.gz
    │   ├── 2015-11-05-22.json.gz
    │   ├── 2015-11-05-23.json.gz
    │   ├── 2015-11-05-3.json.gz
    │   ├── 2015-11-05-4.json.gz
    │   ├── 2015-11-05-5.json.gz
    │   ├── 2015-11-05-6.json.gz
    │   ├── 2015-11-05-7.json.gz
    │   ├── 2015-11-05-8.json.gz
    │   ├── 2015-11-05-9.json.gz
    │   ├── 2015-11-06-0.json.gz
    │   ├── 2015-11-06-1.json.gz
    │   ├── 2015-11-06-10.json.gz
    │   ├── 2015-11-06-11.json.gz
    │   ├── 2015-11-06-12.json.gz
    │   ├── 2015-11-06-13.json.gz
    │   ├── 2015-11-06-14.json.gz
    │   ├── 2015-11-06-15.json.gz
    │   ├── 2015-11-06-16.json.gz
    │   ├── 2015-11-06-17.json.gz
    │   ├── 2015-11-06-18.json.gz
    │   ├── 2015-11-06-19.json.gz
    │   ├── 2015-11-06-2.json.gz
    │   ├── 2015-11-06-20.json.gz
    │   ├── 2015-11-06-21.json.gz
    │   ├── 2015-11-06-22.json.gz
    │   ├── 2015-11-06-23.json.gz
    │   ├── 2015-11-06-3.json.gz
    │   ├── 2015-11-06-4.json.gz
    │   ├── 2015-11-06-5.json.gz
    │   ├── 2015-11-06-6.json.gz
    │   ├── 2015-11-06-7.json.gz
    │   ├── 2015-11-06-8.json.gz
    │   ├── 2015-11-06-9.json.gz
    │   ├── 2015-11-07-0.json.gz
    │   ├── 2015-11-07-1.json.gz
    │   ├── 2015-11-07-10.json.gz
    │   ├── 2015-11-07-11.json.gz
    │   ├── 2015-11-07-12.json.gz
    │   ├── 2015-11-07-13.json.gz
    │   ├── 2015-11-07-14.json.gz
    │   ├── 2015-11-07-15.json.gz
    │   ├── 2015-11-07-16.json.gz
    │   ├── 2015-11-07-17.json.gz
    │   ├── 2015-11-07-18.json.gz
    │   ├── 2015-11-07-19.json.gz
    │   ├── 2015-11-07-2.json.gz
    │   ├── 2015-11-07-20.json.gz
    │   ├── 2015-11-07-21.json.gz
    │   ├── 2015-11-07-22.json.gz
    │   ├── 2015-11-07-23.json.gz
    │   ├── 2015-11-07-3.json.gz
    │   ├── 2015-11-07-4.json.gz
    │   ├── 2015-11-07-5.json.gz
    │   ├── 2015-11-07-6.json.gz
    │   ├── 2015-11-07-7.json.gz
    │   ├── 2015-11-07-8.json.gz
    │   ├── 2015-11-07-9.json.gz
    │   ├── 2015-11-08-0.json.gz
    │   ├── 2015-11-08-1.json.gz
    │   ├── 2015-11-08-10.json.gz
    │   ├── 2015-11-08-11.json.gz
    │   ├── 2015-11-08-12.json.gz
    │   ├── 2015-11-08-13.json.gz
    │   ├── 2015-11-08-14.json.gz
    │   ├── 2015-11-08-15.json.gz
    │   ├── 2015-11-08-16.json.gz
    │   ├── 2015-11-08-17.json.gz
    │   ├── 2015-11-08-18.json.gz
    │   ├── 2015-11-08-19.json.gz
    │   ├── 2015-11-08-2.json.gz
    │   ├── 2015-11-08-20.json.gz
    │   ├── 2015-11-08-21.json.gz
    │   ├── 2015-11-08-22.json.gz
    │   ├── 2015-11-08-23.json.gz
    │   ├── 2015-11-08-3.json.gz
    │   ├── 2015-11-08-4.json.gz
    │   ├── 2015-11-08-5.json.gz
    │   ├── 2015-11-08-6.json.gz
    │   ├── 2015-11-08-7.json.gz
    │   ├── 2015-11-08-8.json.gz
    │   ├── 2015-11-08-9.json.gz
    │   ├── 2015-11-09-0.json.gz
    │   ├── 2015-11-09-1.json.gz
    │   ├── 2015-11-09-10.json.gz
    │   ├── 2015-11-09-11.json.gz
    │   ├── 2015-11-09-12.json.gz
    │   ├── 2015-11-09-13.json.gz
    │   ├── 2015-11-09-14.json.gz
    │   ├── 2015-11-09-15.json.gz
    │   ├── 2015-11-09-16.json.gz
    │   ├── 2015-11-09-17.json.gz
    │   ├── 2015-11-09-18.json.gz
    │   ├── 2015-11-09-19.json.gz
    │   ├── 2015-11-09-2.json.gz
    │   ├── 2015-11-09-20.json.gz
    │   ├── 2015-11-09-21.json.gz
    │   ├── 2015-11-09-22.json.gz
    │   ├── 2015-11-09-23.json.gz
    │   ├── 2015-11-09-3.json.gz
    │   ├── 2015-11-09-4.json.gz
    │   ├── 2015-11-09-5.json.gz
    │   ├── 2015-11-09-6.json.gz
    │   ├── 2015-11-09-7.json.gz
    │   ├── 2015-11-09-8.json.gz
    │   ├── 2015-11-09-9.json.gz
    │   ├── 2015-11-10-0.json.gz
    │   ├── 2015-11-10-1.json.gz
    │   ├── 2015-11-10-10.json.gz
    │   ├── 2015-11-10-11.json.gz
    │   ├── 2015-11-10-12.json.gz
    │   ├── 2015-11-10-13.json.gz
    │   ├── 2015-11-10-14.json.gz
    │   ├── 2015-11-10-15.json.gz
    │   ├── 2015-11-10-16.json.gz
    │   ├── 2015-11-10-17.json.gz
    │   ├── 2015-11-10-18.json.gz
    │   ├── 2015-11-10-19.json.gz
    │   ├── 2015-11-10-2.json.gz
    │   ├── 2015-11-10-20.json.gz
    │   ├── 2015-11-10-21.json.gz
    │   ├── 2015-11-10-22.json.gz
    │   ├── 2015-11-10-23.json.gz
    │   ├── 2015-11-10-3.json.gz
    │   ├── 2015-11-10-4.json.gz
    │   ├── 2015-11-10-5.json.gz
    │   ├── 2015-11-10-6.json.gz
    │   ├── 2015-11-10-7.json.gz
    │   ├── 2015-11-10-8.json.gz
    │   ├── 2015-11-10-9.json.gz
    │   ├── 2015-11-11-0.json.gz
    │   ├── 2015-11-11-1.json.gz
    │   ├── 2015-11-11-10.json.gz
    │   ├── 2015-11-11-11.json.gz
    │   ├── 2015-11-11-12.json.gz
    │   ├── 2015-11-11-13.json.gz
    │   ├── 2015-11-11-14.json.gz
    │   ├── 2015-11-11-15.json.gz
    │   ├── 2015-11-11-16.json.gz
    │   ├── 2015-11-11-17.json.gz
    │   ├── 2015-11-11-18.json.gz
    │   ├── 2015-11-11-19.json.gz
    │   ├── 2015-11-11-2.json.gz
    │   ├── 2015-11-11-20.json.gz
    │   ├── 2015-11-11-21.json.gz
    │   ├── 2015-11-11-22.json.gz
    │   ├── 2015-11-11-23.json.gz
    │   ├── 2015-11-11-3.json.gz
    │   ├── 2015-11-11-4.json.gz
    │   ├── 2015-11-11-5.json.gz
    │   ├── 2015-11-11-6.json.gz
    │   ├── 2015-11-11-7.json.gz
    │   ├── 2015-11-11-8.json.gz
    │   ├── 2015-11-11-9.json.gz
    │   ├── 2015-11-12-0.json.gz
    │   ├── 2015-11-12-1.json.gz
    │   ├── 2015-11-12-10.json.gz
    │   ├── 2015-11-12-11.json.gz
    │   ├── 2015-11-12-12.json.gz
    │   ├── 2015-11-12-13.json.gz
    │   ├── 2015-11-12-14.json.gz
    │   ├── 2015-11-12-15.json.gz
    │   ├── 2015-11-12-16.json.gz
    │   ├── 2015-11-12-17.json.gz
    │   ├── 2015-11-12-18.json.gz
    │   ├── 2015-11-12-19.json.gz
    │   ├── 2015-11-12-2.json.gz
    │   ├── 2015-11-12-20.json.gz
    │   ├── 2015-11-12-21.json.gz
    │   ├── 2015-11-12-22.json.gz
    │   ├── 2015-11-12-23.json.gz
    │   ├── 2015-11-12-3.json.gz
    │   ├── 2015-11-12-4.json.gz
    │   ├── 2015-11-12-5.json.gz
    │   ├── 2015-11-12-6.json.gz
    │   ├── 2015-11-12-7.json.gz
    │   ├── 2015-11-12-8.json.gz
    │   ├── 2015-11-12-9.json.gz
    │   ├── 2015-11-13-0.json.gz
    │   ├── 2015-11-13-1.json.gz
    │   ├── 2015-11-13-10.json.gz
    │   ├── 2015-11-13-11.json.gz
    │   ├── 2015-11-13-12.json.gz
    │   ├── 2015-11-13-13.json.gz
    │   ├── 2015-11-13-14.json.gz
    │   ├── 2015-11-13-15.json.gz
    │   ├── 2015-11-13-16.json.gz
    │   ├── 2015-11-13-17.json.gz
    │   ├── 2015-11-13-18.json.gz
    │   ├── 2015-11-13-19.json.gz
    │   ├── 2015-11-13-2.json.gz
    │   ├── 2015-11-13-20.json.gz
    │   ├── 2015-11-13-21.json.gz
    │   ├── 2015-11-13-22.json.gz
    │   ├── 2015-11-13-23.json.gz
    │   ├── 2015-11-13-3.json.gz
    │   ├── 2015-11-13-4.json.gz
    │   ├── 2015-11-13-5.json.gz
    │   ├── 2015-11-13-6.json.gz
    │   ├── 2015-11-13-7.json.gz
    │   ├── 2015-11-13-8.json.gz
    │   ├── 2015-11-13-9.json.gz
    │   ├── 2015-11-14-0.json.gz
    │   ├── 2015-11-14-1.json.gz
    │   ├── 2015-11-14-10.json.gz
    │   ├── 2015-11-14-11.json.gz
    │   ├── 2015-11-14-12.json.gz
    │   ├── 2015-11-14-13.json.gz
    │   ├── 2015-11-14-14.json.gz
    │   ├── 2015-11-14-15.json.gz
    │   ├── 2015-11-14-16.json.gz
    │   ├── 2015-11-14-17.json.gz
    │   ├── 2015-11-14-18.json.gz
    │   ├── 2015-11-14-19.json.gz
    │   ├── 2015-11-14-2.json.gz
    │   ├── 2015-11-14-20.json.gz
    │   ├── 2015-11-14-21.json.gz
    │   ├── 2015-11-14-22.json.gz
    │   ├── 2015-11-14-23.json.gz
    │   ├── 2015-11-14-3.json.gz
    │   ├── 2015-11-14-4.json.gz
    │   ├── 2015-11-14-5.json.gz
    │   ├── 2015-11-14-6.json.gz
    │   ├── 2015-11-14-7.json.gz
    │   ├── 2015-11-14-8.json.gz
    │   ├── 2015-11-14-9.json.gz
    │   ├── 2015-11-15-0.json.gz
    │   ├── 2015-11-15-1.json.gz
    │   ├── 2015-11-15-10.json.gz
    │   ├── 2015-11-15-11.json.gz
    │   ├── 2015-11-15-12.json.gz
    │   ├── 2015-11-15-13.json.gz
    │   ├── 2015-11-15-14.json.gz
    │   ├── 2015-11-15-15.json.gz
    │   ├── 2015-11-15-16.json.gz
    │   ├── 2015-11-15-17.json.gz
    │   ├── 2015-11-15-18.json.gz
    │   ├── 2015-11-15-19.json.gz
    │   ├── 2015-11-15-2.json.gz
    │   ├── 2015-11-15-20.json.gz
    │   ├── 2015-11-15-21.json.gz
    │   ├── 2015-11-15-22.json.gz
    │   ├── 2015-11-15-23.json.gz
    │   ├── 2015-11-15-3.json.gz
    │   ├── 2015-11-15-4.json.gz
    │   ├── 2015-11-15-5.json.gz
    │   ├── 2015-11-15-6.json.gz
    │   ├── 2015-11-15-7.json.gz
    │   ├── 2015-11-15-8.json.gz
    │   └── 2015-11-15-9.json.gz
    └── taxi
        ├── yellow_tripdata_2015-11.parquet
        └── yellow_tripdata_2015-12.parquet

3 directories, 362 files
```

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]

File: java/bench/README.md (hunk 10 above)
Comment (mwlon) [reply java/bench/README.md:27]: Ah you're right, I had downsized the github data. I'll update to 4MB

Comment (dongjoon-hyun) [reply java/bench/README.md]: Thanks!

**Comment by dongjoon-hyun** [comment]
I'll resume the verification now.

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Unfortunately, I cannot reproduce your result in the PR description. The root cause of failure was due to ORC-1109.

```
$ java -jar hive/target/orc-benchmarks-hive-1.9.0-SNAPSHOT-uber.jar decimal data
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.openjdk.jmh.util.Utils (file:/Users/dongjoon/APACHE/orc-merge/java/bench/hive/target/orc-benchmarks-hive-1.9.0-SNAPSHOT-uber.jar) to field java.io.Console.cs
WARNING: Please consider reporting this to the maintainers of org.openjdk.jmh.util.Utils
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
# JMH version: 1.20
# VM version: JDK 11.0.15, VM 11.0.15+0
# VM invoker: /opt/homebrew/Cellar/openjdk@11/11.0.15/libexec/openjdk.jdk/Contents/Home/bin/java
# VM options: -server -Xms256m -Xmx2g -Dbench.root.dir=/Users/dongjoon/DATA/orc
# Warmup: 2 iterations, 10 s each
# Measurement: 5 iterations, 10 s each
# Timeout: 10 min per iteration
# Threads: 1 thread, will synchronize iterations
# Benchmark mode: Average time, time/op
# Benchmark: org.apache.orc.bench.hive.DecimalBench.read
# Parameters: (version = ORIGINAL)

# Run progress: 0.00% complete, ETA 00:04:40
# Fork: 1 of 1
# Warmup Iteration   1: [WARN ] Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
<failure>

java.io.FileNotFoundException: File /Users/dongjoon/DATA/orc/generated/taxi/orc.none does not exist
	at org.apache.hadoop.fs.RawLocalFileSystem.deprecatedGetFileStatus(RawLocalFileSystem.java:779)
```

**Review by dongjoon-hyun — APPROVED** [review]
Let me merge this first and fix the issue as another issue. Thank you for your patience and contribution, @mwlon .

+1, LGTM (with the known issue which is irrelevant to this contribution).

**Comment by dongjoon-hyun** [comment]
Merged to main/1.8/1.7.

**Comment by dongjoon-hyun** [comment]
Here is the PR for `orc.none`-related failure.
- https://github.com/apache/orc/pull/1144
